File: mp2i/files/C7/tri_rapide.ml
let rec separe l pivot =
  match l with 
  | [] -> [], []
  | h::t -> let l1,l2 = separe t pivot in if h<pivot then h::l1,l2 else 
    if h>pivot then l1,h::l2 else l1,l2;;

let rec tri_rapide l =
  match l with
  | [] -> []
  | [x] -> [x]
  | h::t ->  let l1,l2 = separe l h in
  let lt1 = tri_rapide l1 in
  let lt2 = tri_rapide l2 in
  lt1 @ (h::lt2);;

File: itc/files/C8/tri_rapide.py
def partitionne(l):
    pivot = l[-1]
    plus_petit = [x for x in l if x<pivot]
    plus_grand = [x for x in l if x>pivot]
    return plus_petit, plus_grand

def tri_rapide(l):
    if len(l) <= 1:
        return l
    l1,l2 = partitionne(l)
    lt1 = tri_rapide(l1)
    lt2 = tri_rapide(l2)
    return lt1+[l[-1]]+lt2

l = [3, 7, 1, 9, 5, 2, 6, 4, 8]
lt = tri_rapide(l)
print(lt)